services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/TableSchema.java

```diff
@@ -237,11 +238,25 @@ default T mapToItem(Map<String, AttributeValue> attributeMap, boolean preserveEm
      * SDK can work with.
      *
      * @param item The modelled Java object to convert into a map of attributes.
-     * @param ignoreNulls If set to true; any null values in the Java object will not be added to the output map.
-     *                    If set to false; null values in the Java object will be added as {@link AttributeValue} of
-     *                    type 'nul' to the output map.
+     * @param configuration configuration options for mapping an object to a map with attribute values
      * @return A map of String to {@link AttributeValue} representing all the modelled attributes in the model object.
      */
+    default Map<String, AttributeValue> itemToMap(T item,
+                                          MappingConfiguration configuration) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * This is deprecated in favour of itemToMap(T item, MappingConfiguration
+     * configuration) This method takes a modelled object and converts it into a raw map of
+     * {@link AttributeValue} that the DynamoDb low-level SDK can work with.
+     *
+     * @param item        The modelled Java object to convert into a map of attributes.
+     * @param ignoreNulls If set to true; any null values in the Java object will not be added to the output map. If set to false;
+     *                    null values in the Java object will be added as {@link AttributeValue} of type 'nul' to the output map.
+     * @return A map of String to {@link AttributeValue} representing all the modelled attributes in the model object.
+     */
+    @Deprecated
```

Comment:
Can we add `@deprecated` to the javadoc? Example: https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/retry/RetryPolicy.java#L46

Reply:
done